docs/project/development-process/tags-maintenance-branches-in-our-code.md

```diff
@@ -89,6 +89,27 @@ The upstream repository doesn't contain the maintenance branches that are the ba
 - When our branch exactly matches the contents of the hotfix's tarball, tag it as `vUPSTREAMVERSION-xcpng`. This tag will be used to checkout for the next maintenance update, and as the base reference to generate patches for our RPM.
 - At last, we can apply our specific commits on top: rebase or cherry-pick them from the previous maintenance branch for this release (example: `2.30.3-8.2`) and resolve conflicts carefully. Of course, drop patches that were already merged upstream.
 
+### Special case: `xapi`
+
+The primary objective is to do everything necessary to **push our patches upstream**. We
+can maintain patches in our branches so that they are available in our RPMs as early as possible,
+but it's crucial that these patches ultimately be awaiting upstream acceptance.
+
+#### How to manage patches?
+
+- Create a new `UPSTREAMVERSION-XCPNGVERSION` branch (example: `v25.6.0-8.3`).
+- To update either the upstream version or the xcpng version, rebase or cherry-pick your commits
+from the current maintenance branch for the previous release (e.g, `v24.19.2-8.2`), and
+resolve conflicts carefully. Reorganize commits as needed to maintain a consistent history.
+Of course, drop any patches that have already been merged upstream.
+- For example, if the upstream version changes:
+  - create a new branch from the given tag: `git switch -c v25.6.0-8.3 v25.6.0`
+  - cherry-pick the patches from the older version.
```

Comment:
I think rebase is generally preferrable over cherry-pick, but whatever works best in practice...

```suggestion
  - rebase or cherry-pick the patches from the older version.
```